base/expr.jl

```diff
@@ -582,7 +582,7 @@ macro generated(f)
                                    # https://github.com/JuliaLang/julia/issues/25678
                                    Expr(:block,
                                         :(local tmp = $body),
```

Comment:
Looking at this again, this also shouldn't use `tmp` as a name without any escaping, since this example currently errors:
```julia
julia> @test @eval Module() begin
           @generated f(tmp) = tmp
           f(1) == Int
       end
Error During Test at REPL[4]:1
  Test threw exception
  Expression: #= REPL[4]:1 =# @eval Module() begin
        #= REPL[4]:2 =# @generated f(tmp) = begin
                    tmp
                end
        f(1) == Int
    end
  syntax: local variable name "tmp" conflicts with an argument
[...]
```
We should either escape only the relevant parts properly or use `gensym` here. (This change to using `GlobalRef` here still seems useful regardless.)